sql/sql_expression.h

```diff
@@ -504,23 +504,31 @@ struct GetAllColumnsOutput {
 
 struct ColumnFilter {
     typedef std::function<ColumnName (const ColumnName &)> Exec;
+    bool init;
    
-    ColumnFilter() 
+    ColumnFilter() : init(false)
     {
-
     }    
 
-    ColumnFilter(Exec exec)
-        : exec(std::move(exec))
+    ColumnFilter(Exec exec_)
+        : exec(std::move(exec_))
     {
+        init = true;
+    }
+
+    static ColumnFilter identity() {
```

Comment:
Not sure I understand how one way or the other makes it easier to know that the filter does nothing.